blog/content/post/2022-01-03-pcie_screamer.md

```diff
@@ -111,108 +108,194 @@ To prepare `enjoy-digital/pcie_screamer`, the following steps were tried:
+mgabryelski@maciej-HP:~$ lspci
+00:00.0 Host bridge: Intel Corporation Xeon E3-1200 v2/3rd Gen Core processor DRAM Controller (rev 09)
+00:02.0 VGA compatible controller: Intel Corporation IvyBridge GT2 [HD Graphics 4000] (rev 09)
+. . .
+00:1f.5 IDE interface: Intel Corporation 7 Series/C210 Series Chipset Family 2-port SATA Controller [IDE mode] (rev 04)
+02:00.0 Memory controller: Xilinx Corporation Device 7022 (rev 01)
+mgabryelski@maciej-HP:~$
+```
+As you can see the last device was: **Memory controller: Xilinx Corporation Device
+7022 (rev 01)** and this is how PCIScreamer should be visible on PCI bus.
+After connecting the computer via USB 3.0 cable to the connector on the PCIScreamer
+board, it should be visible in the OS. In order to check this, we ran the command 
+in the console:
+```bash
+   $ lsusb
+```
+The result are below:
+```bash
+mgabryelski@maciej-HP:~$ lsusb
+Bus 002 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
+Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+Bus 001 Device 004: ID 093a:2510 Pixart Imaging, Inc. Optical Mouse
+Bus 001 Device 003: ID 04d9:1602 Holtek Semiconductor, Inc. USB Keyboard
+Bus 001 Device 005: ID 0403:601f Future Technology Devices International, Ltd Myriad-RF LimeSDR-Mini
+Bus 001 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
+Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
+Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+mgabryelski@maciej-HP:~$ 
+```
+The device: **Bus 001 Device 005: ID 0403:601f Future Technology Devices International, Ltd Myriad-RF LimeSDR-Mini** is FTDI (USB 3.0), and this is also 
+valid.
+
+Now we can go to tests with the application `PCIleech`. We can download ready-made
+binary packages (Linux x64) from this page [PCIleech software](https://github.com/ufrisk/pcileech/releases/tag/v4.15) Po rozpakowaniu archiwum z aplikacją wydajemy w konsoli komendę:
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ ./pcileech
+./pcileech: error while loading shared libraries: libcrypto.so.1.1: cannot open shared object file: No such file or directory
+mgabryelski@maciej-HP:~/PCIScreamer$
+```
+It turns out that `pcileech` depends on the archival version of the library
+`openssl` (which is not available as package in OS repository). The solution was
+building this library from sources. Here is sequence of command issued in console
+in order to build this library:
+```bash
+wget https://www.openssl.org/source/openssl-1.1.1o.tar.gz
+cd openssl-1.1.1o
+./config
+make
+make test
+sudo make install
+find / -name libssl.so.1.1
+ln -s /usr/local/lib64/libssl.so.1.1  /usr/lib64/libssl.so.1.1
+ln -s /usr/local/lib64/libssl.so.1.1  /usr/lib/libssl.so.1.1
+find / -name libcrypto.so.1.1
+ln -s /home/ubuntu/openssl-1.1.1o/libcrypto.so.1.1    /usr/lib64/libcrypto.so.1.1
+ln -s /home/ubuntu/openssl-1.1.1o/libcrypto.so.1.1     /usr/lib/libcrypto.so.1.1
```

Comment:
This is wrong.
First, we should never create files /usr ourselves (vs /usr/local). Generally, `/usr` is managed by system's package manager, e.g. apt and dpkg, manually modifying system files may result in mayor problems.
Here we override system's provided OpenSSL with an older version, which may have critical vulnerabilities or may not work at all with modern applications that were built agains newer OpenSSL version.
In Ubuntu OpenSSL is installed at `/usr/lib/x86_64-linux-gnu/libcrypto.so.1.1`. Since binary worked after installation I assume that `/usr/lib` has precedence over `/usr/lib/x86_64-linux-gnu/` and that OpenSSL got overrided globally which may put system at risk.

OpenSSL can be installed to `/usr/local` (or anywhere else) and we may point to its library directory using `LD_LIBRARY_PATH` env variable, also there is possibility to use `patchelf` to override library path in executable itself and avoid using `LD_LIBRARY_PATH`.

However, if `pcileech` can be easily built we should prefer that instead of providing old libraries.